modules/structs/activity_visibility.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package structs
+
+// ActivityVisibility defines the activities shown
+type ActivityVisibility int
+
+const (
+	// ActivityVisibilityPublic show public activities
+	ActivityVisibilityPublic ActivityVisibility = iota
+
+	// ActivityVisibilityAll shows all activities
+	ActivityVisibilityAll
+
+	// ActivityVisibilityNone show no activities
+	ActivityVisibilityNone
+)
+
+// ActivityVisibilities is a map of ActivityVisibility types
+var ActivityVisibilities = map[string]ActivityVisibility{
+	"public": ActivityVisibilityPublic,
+	"all":    ActivityVisibilityAll,
+	"none":   ActivityVisibilityNone,
+}
+
+// ShowPublic returns true if ActivityVisibility is public
+func (vt ActivityVisibility) ShowPublic() bool {
+	return vt == ActivityVisibilityPublic
+}
+
+// ShowAll returns true if ActivityVisibility is all
+func (vt ActivityVisibility) ShowAll() bool {
+	return vt == ActivityVisibilityAll
+}
+
+// ShowNone returns true if ActivityVisibility is none
+func (vt ActivityVisibility) ShowNone() bool {
+	return vt == ActivityVisibilityNone
+}
+
+// String provides the mode string of the visibility type (public, all, none)
+func (vt ActivityVisibility) String() string {
```

Comment:
Is this "String" method really used? I think it only causes problems. For example: it's impossible to render the value directly into template.

So IMO either:

* Remove such "String" method, and then just output the int value into template directly
* Or if it is really needed, name it to "ToString" or something else, do not make it a "Stringer"